openpype/hosts/blender/api/plugin.py

```diff
@@ -49,6 +69,20 @@ def get_unique_number(
     return f"{count:0>2}"
 
 
+def get_model_unique_number(current_container_name:str) -> str:
+    """Return a unique number based on the asset name."""
+    data_collections = bpy.data.collections
+    container_names = [c.name for c in data_collections if c.get(AVALON_PROPERTY)]
+    if container_names == []:
+        return "001"
+    count = 1
+    name = f"{current_container_name}_{count:0>3}"
+    while name in container_names:
```

Comment:
Why do you use a `while` instead of a `for` here?

Reply:
It's not my function but they increment the name as long as it's in container_names. If it's not inside it's the right increment